src/MetadataManager/Metadata.ts

```diff
@@ -0,0 +1,189 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as flatbuffers from 'flatbuffers';
+import * as fs from 'fs';
+import * as vscode from 'vscode';
+
+import {ToolchainInfo} from '../Backend/Toolchain';
+import * as Circle from '../CircleEditor/circle_schema_generated';
+import {obtainWorkspaceRoot} from '../Utils/Helpers';
+
+import {isOneExplorerTargetFile} from '../Utils/Helpers';
+
+type BuildInfoKeys = 'onecc'|'toolchain'|'cfg';
+
+interface BuildInfoObj {
+  'onecc': string, 'toolchain': ToolchainInfo|undefined, 'cfg': any
+}
+
+export class BuildInfo {
+  private static _map = new Map<string, BuildInfoObj>();
+
+  public static set(path: string, key: BuildInfoKeys, value: any) {
+    const relPath = vscode.workspace.asRelativePath(path);
+    let info = BuildInfo._map.get(relPath);
+    if (info === undefined) {
+      info = {onecc: '', toolchain: undefined, cfg: undefined};
+      BuildInfo._map.set(relPath, info);
+    }
+    info[key] = value;
+  }
+
+  public static save(metaEntry: any, uri: vscode.Uri) {
+    const path = vscode.workspace.asRelativePath(uri);
+    const info = BuildInfo._map.get(path);
+    if (info) {
+      metaEntry['onecc-version'] = info['onecc'];
+      metaEntry['toolchain-version'] = info['toolchain'] ?.version ?.str();
+      metaEntry['cfg-settings'] = info['cfg'];
+    }
+    BuildInfo._map.delete(path);
+  }
+}
+
+export class Operator {
+  public static async get(uri: vscode.Uri) {
+    if (!uri.fsPath.endsWith('.circle')) {
+      return [];
+    }
+    const bytes = new Uint8Array(await vscode.workspace.fs.readFile(uri));
+    const buf = new flatbuffers.ByteBuffer(bytes);
+    const model = Circle.Model.getRootAsModel(buf).unpack();
+    const operators = model.operatorCodes.map((operator) => {
+      Circle.BuiltinOperator[operator.deprecatedBuiltinCode];
+    });
+    return operators;
+  }
+}
+
+export class Metadata {
+  constructor() {}
+
+  public static async getObj(hash: string) {
+    if (!vscode.workspace.workspaceFolders) {
+      return;
+    }
+
+    const jsonUri = vscode.Uri.joinPath(
+        vscode.workspace.workspaceFolders[0].uri,
+        `.meta/hash_objects/${hash.substring(0, 2)}/${hash.substring(2)}.json`);
+
+    if (fs.existsSync(jsonUri.fsPath)) {
+      return JSON.parse(Buffer.from(await vscode.workspace.fs.readFile(jsonUri)).toString());
+    }
+    // TODO: Error Handling
+    return undefined;
+  }
+
+  public static async setObj(hash: string, obj: object) {
+    const workspaceroot = obtainWorkspaceRoot();
+    const jsonUri = vscode.Uri.joinPath(
+        vscode.Uri.file(workspaceroot),
+        `.meta/hash_objects/${hash.substring(0, 2)}/${hash.substring(2)}.json`);
```

Comment:
This may require a big change, but could you rename the directory..? `.meta/` -> `.one-vscode/info/`

Reply:
We will apply the comments. I think replacing `.meta/` -> `.one-vscode/info/` will be not that hard.

Reply:
Thanks :-D